src/main/java/com/societegenerale/commons/plugin/rules/DontReturnNullCollectionTest.java

```diff
@@ -55,9 +52,10 @@ public boolean apply(JavaMethod input) {
         @Override
         public boolean apply(JavaMethod input) {
 
-          Class returnedClass = input.getReturnType().toErasure().reflect();
+          JavaClass returnedJavaClass = input.getReturnType().toErasure();
 
-          return returnedClass.equals(List.class) || returnedClass.equals(Set.class) ;
+          return returnedJavaClass.getAllRawInterfaces().stream()
+              .anyMatch(implementedInterface -> implementedInterface.reflect().equals(Collection.class));
```

Comment:
Using [javaClass.isEquivalentTo](https://javadoc.io/doc/com.tngtech.archunit/archunit/latest/com/tngtech/archunit/core/domain/JavaClass.html#isEquivalentTo(java.lang.Class)), you can avoid calling `reflect`.